include/seqan3/search/search_result.hpp

```diff
@@ -28,20 +29,45 @@ struct policy_result_builder;
 namespace seqan3
 {
 
-/*!\brief A data holder for the search result.
+/*!\brief The result class generated by the seqan3::seach algorithm.
  * \ingroup search
- * \tparam query_id_type The type of the query_id.
- * \tparam cursor_type The type of the cursor.
- * \tparam reference_id_type The type of the reference_id.
- * \tparam reference_begin_pos_type The type of the reference_begin_pos.
+ * \tparam query_id_type The type of the query_id; must model std::integral or be the empty_type.
+ * \tparam cursor_type The type of the cursor; must model seqan3::fm_index_cursor_specialisation or be the empty_type.
+ * \tparam reference_id_type The type of the reference_id; must model std::integral or be the empty_type.
+ * \tparam reference_begin_pos_type The type of the reference_begin_pos; must model std::integral or be the empty_type.
  *
  * \if DEV
- * \note If the information is not available the type of the respective data is of seqan3::detail::empty_type.
+ * \note If the information is not available the type of the respective data field defaults to seqan3::detail::empty_type.
  * \endif
+ *
+ * The seqan3::search algorithm returns a range of hits. A single hit is stored in a seqan3::search_result.
+ * By default, the search result contains the query id, the reference id where the query matched and the begin
+ * position in the reference where the query sequence starts to match the reference sequence.
+ * Those information can be accessed via the respective member functions.
+ *
+ * The following member functions exist:
+ *
+ * * seqan3::search_result::query_id()
+ * * seqan3::search_result::index_cursor()
+ * * seqan3::search_result::reference_id()
+ * * seqan3::search_result::reference_begin_pos()
+ *
+ * Note that the index cursor is not included in a hit by default. If you are trying to access that information,
+ * an exception will be thrown. You con configure the result of the search with the output configuration
+ * (see seqan3::search_cfg::output).
  */
 template <typename query_id_type, typename cursor_type, typename reference_id_type, typename reference_begin_pos_type>
 class search_result
 {
+    static_assert(std::integral<query_id_type> || std::same_as<query_id_type, detail::empty_type>,
+                  "The query_id_type must either model std::integral or be detail::empty_type.");
```

Comment:
Why can't this be simply template constraints above? What is the added value of these static asserts. It just writes out what it tests. I don't see this as a justification for making them static assert rather than class constraints?

Reply:
I'm pretty sure agreed that we will use static asserts if the requires do not participate in overload resolution but I cannot find it in the core meeting notes.

Reply:
I really don't see any benefit in these static asserts. Once we have support for the types in doxygen we can have a proper documentation. I really think that a static assert helps to identify hard to catch invariants. But I am in srong favour of constraining the primary templates for class templates, function templates and variable templates. Who knows when and who will overload it.

Reply:
I changed it